hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -91,5 +97,114 @@ func SetupWebhookWithManager(mgr ctrl.Manager) error {
+	hc, ok := obj.(*hyperv1.HostedCluster)
+	if !ok {
+		return fmt.Errorf("wrong type %T for validation, instead of HostedCluster", obj)
+	}
+
+	switch hc.Spec.Platform.Type {
+	case hyperv1.KubevirtPlatform:
+		return v.validateCreateKubevirtHostedCluster(ctx, hc)
+	default:
+		return nil // no validation needed
+	}
+}
+
+func (v hostedClusterValidator) ValidateUpdate(_ context.Context, _, _ runtime.Object) error {
+	return nil
+}
+
+func (v hostedClusterValidator) ValidateDelete(_ context.Context, _ runtime.Object) error {
+	return nil
+}
+
+func (v hostedClusterValidator) validateCreateKubevirtHostedCluster(ctx context.Context, hc *hyperv1.HostedCluster) error {
+	if hc.Spec.Platform.Kubevirt == nil {
+		return fmt.Errorf("the spec.platform.kubevirt field is missing in the HostedCluster resource")
+	}
+
+	return kvValidator.validate(ctx, v.client, hc)
+}
+
+type nodePoolValidator struct {
+	client client.Client
+}
+
+func newNodePoolValidator(client client.Client) *nodePoolValidator {
+	return &nodePoolValidator{
+		client: client,
+	}
+}
+
+func (v nodePoolValidator) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	np, ok := obj.(*hyperv1.NodePool)
+	if !ok {
+		return fmt.Errorf("wrong type %T for validation, instead of NodePool", obj)
+	}
+
+	switch np.Spec.Platform.Type {
+	case hyperv1.KubevirtPlatform:
+		return v.validateCreateKubevirtNodePool(ctx, np)
+	default:
+		return nil // no validation needed
+	}
+}
+
+func (v nodePoolValidator) ValidateUpdate(_ context.Context, _, _ runtime.Object) error {
+	return nil
+}
+
+func (v nodePoolValidator) ValidateDelete(_ context.Context, _ runtime.Object) error {
+	return nil
+}
+
+func (v nodePoolValidator) validateCreateKubevirtNodePool(ctx context.Context, np *hyperv1.NodePool) error {
+	hc := &hyperv1.HostedCluster{}
+	err := v.client.Get(ctx, client.ObjectKey{Name: np.Spec.ClusterName, Namespace: np.Namespace}, hc)
+	if err != nil {
+		return fmt.Errorf("failed to retrive HostedCluster %s/%s; %w", np.Namespace, np.Spec.ClusterName, err)
+	}
+
+	return kvValidator.validate(ctx, v.client, hc)
+}
+
+type kubevirtClusterValidator struct {
+	clientMap kubevirtexternalinfra.KubevirtInfraClientMap
+}
+
+func (v kubevirtClusterValidator) validate(ctx context.Context, cli client.Client, hc *hyperv1.HostedCluster) error {
+	if hc.Spec.Platform.Kubevirt == nil {
+		return fmt.Errorf("the spec.platform.kubevirt field is missing in the HostedCluster resource")
+	}
+
+	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hc.Namespace, hc.Name).Name
+	cl, err := v.clientMap.DiscoverKubevirtClusterClient(ctx, cli, hc.Spec.InfraID, hc.Spec.Platform.Kubevirt.Credentials, controlPlaneNamespace, hc.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to connect external infra cluster; %w", err)
+	}
+
+	if _, isTimeout := ctx.Deadline(); !isTimeout {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, time.Second*3)
```

Comment:
i'd feel better if we extended this to a 5 second timeout... it likely doesn't matter, but i'm more on the conservative side when it comes to timing out operations.

Reply:
I'll make it 10, to be closer to the webhook T/O (15 sec)